src/grid/Grid.tsx

```diff
@@ -113,6 +118,27 @@ function offset(element: HTMLElement, relativeTo: Element): Offset {
   }
 }
 
+interface LayoutContext {
+  setGeneration: Dispatch<SetStateAction<number | null>>;
+}
+
+const LayoutContext = createContext<LayoutContext | null>(null);
+
+/**
+ * Enables Grid to react to layout changes. You must call this in your Layout
+ * component or else Grid will not be reactive.
+ */
+export function useLayout(): void {
```

Comment:
why does this get a use name?
It seems like it behaves more like a schedule recompute method?

Reply:
It gets a `use` name because it is a custom React hook (it calls `useContext` and `useEffect` internally). Do you think something other than `useLayout` would better communicate what it does though?

Reply:
`useLayoutEffect` maybe?

Reply:
That's the name of a built-in React hook, so it'd probably be confusing, but I went with `useUpdateLayout` which is hopefully an improvement